public/js/grid/timetable_util.js

```diff
@@ -121,12 +121,16 @@ function restoreFromCookies() {
     var selectedCourseCookie = getCookie('selected-courses');
     var selectedSectionCookie = getCookie('selected-lectures');
 
-    if (selectedCourseCookie.length === 0) {
-        selectedCourseCookie = [];
+    if (selectedCourseCookie === undefined || selectedCourseCookie.length === 0) {
+        selectedCourseCookie = '[]';
+    } else {
+        selectedCourseCookie = '["' + selectedCourseCookie.split('_').join('","') + '"]';
```

Comment:
How we handle the cookies can be greatly simplified here. It should be possible to recover the list of courses directly just by doing the `split('_')`, and not use the parseJSON function at all.